### PR TITLE
feat(anvil): allow pass headers to `fork-url`

### PR DESCRIPTION
**Pull request by vbrvk**
Closes https://github.com/foundry-rs/foundry/issues/6173

### CODE REVIEW

**Comment by vbrvk** [comment]
image


After added `headers` param clippy does not satisfied with too many arguments. @Evalir what do you think the best option to fix it?

File: crates/common/src/runtime_client.rs [hunk 1]
```diff
@@ -113,6 +117,7 @@ impl RuntimeClient {
         timeout: Duration,
         compute_units_per_second: u64,
         jwt: Option<String>,
+        headers: Vec<String>,
```
Comment (Evalir) [inline crates/common/src/runtime_client.rs:120]: So, just to not get blocked on this let's just ignore the clippy lint for now—however a good followup would be making a `RuntimeClientBuilder` type that can build a `RuntimeClient` so we avoid this (and make the `RuntimeClient` members public).

Comment (vbrvk) [reply crates/common/src/runtime_client.rs]: Fixed with builder

**Review by mattsse — CHANGES_REQUESTED** [review]
please rm all additional `;`

makes it harder to review

**Comment by vbrvk** [comment]
Fixed, actually I was thinking about not adding them, but contributing guide recommends to enable `formatOnSave` option, so I did :)

**Comment by Evalir** [comment]
yep @vbrvk we use nightly clippy/formatting, so the appropiate command should be `cargo +nightly fmt`—we can probably make this a bit clearer in the contribution guide

File: crates/common/src/runtime_client.rs [hunk 2]
```diff
@@ -142,16 +138,28 @@ impl RuntimeClient {
...
+                    let parts: Vec<_> = header.split(':').map(|s| s.trim()).collect();
+
+                    let key = parts.first().expect(&err_message);
+                    let val = parts.get(1).expect(&err_message);
+
+                    headers.insert(
+                        HeaderName::from_lowercase(key.to_lowercase().as_bytes())
+                            .expect(&err_message),
+                        HeaderValue::from_str(val).expect(&err_message),
+                    );
```
Comment (Evalir) [inline crates/common/src/runtime_client.rs:155]: let's avoid `expect`s and instead use `?`

**Review by Evalir — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
lgtm!

ty

**Pull request merged**